test/user.e2e-spec.ts

```diff
@@ -0,0 +1,128 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { INestApplication, ValidationPipe } from '@nestjs/common';
+import * as request from 'supertest';
+import { UserModule } from '../src/api/user/user.module';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../src/api/user/user.entity';
+import { CreateUserDto } from '../src/api/user/user.dto';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+
+describe('Users API Testing (e2e)', () => {
+  let app: INestApplication;
+  let userRepository: Repository<User>;
+  const createUser: CreateUserDto = { username: 'Klaven', isAdmin: false };
+  const result = {
+    id: expect.any(Number),
+    username: 'Klaven',
+    isAdmin: false,
+    createdAt: expect.any(String),
+    updatedAt: expect.any(String),
+    deletedAt: null,
+  };
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        UserModule,
+        TypeOrmModule.forRootAsync({
+          imports: [ConfigModule],
+          useFactory: (config: ConfigService) => ({
+            type: 'postgres',
+            host: config.get('DATABASE_HOST'),
+            port: config.get('DATABASE_PORT'),
+            username: config.get('DATABASE_USER'),
+            password: '',
+            database: process.env.TEST_DATABASE_NAME,
+            entities: [User],
+            synchronize: true,
```

Comment:
probably want to standardize how you reference env vars.  Also it looks like in some places you're using DATABASE and in others TEST_DATABASE. I'd also expect things like type, password, etc. not to be configured in the test. You might also want to look into the `autoloadEntities` option.